pyomo/core/tests/unit/test_set.py

```diff
@@ -6388,3 +6395,254 @@ def test_issue_1112(self):
+        # using concrete model and initialization
+        problem = ConcreteModel()
+        # non-indexed sets not using the within argument
+        problem.A = Set(initialize=[1, 2, 3])
+        problem.B = Set(dimen=2, initialize=[(1, 2), (3, 4), (5, 6)])
+        # non-indexed sets using within argument
+        problem.subset_A = Set(within=problem.A, initialize=[2, 3])
+        problem.subset_B = Set(within=problem.B, dimen=2, initialize=[(1, 2), (5, 6)])
+        # indexed sets not using the within argument
+        problem.C = Set(problem.A, initialize={1: [-1, 3], 2: [4, 7], 3: [3, 8]})
+        problem.D = Set(
+            problem.B, initialize={(1, 2): [1, 5], (3, 4): [3], (5, 6): [6, 8, 9]}
+        )
+        # indexed sets using an indexed set for the within argument
+        problem.subset_C = Set(
+            problem.A, within=problem.C, initialize={1: [-1], 2: [4], 3: [3, 8]}
+        )
+        problem.subset_D = Set(
+            problem.B,
+            within=problem.D,
+            initialize={(1, 2): [1, 5], (3, 4): [], (5, 6): [6]},
+        )
+        # indexed sets using a non-indexed set for the within argument
+        problem.E = Set([0, 1], within=problem.A, initialize={0: [1, 2], 1: [3]})
+        problem.F = Set(
+            [(1, 2, 3), (4, 5, 6)],
+            within=problem.B,
+            initialize={(1, 2, 3): [(1, 2)], (4, 5, 6): [(3, 4)]},
+        )
+        # check them
+        self.assertEqual(list(problem.A), [1, 2, 3])
+        self.assertEqual(list(problem.B), [(1, 2), (3, 4), (5, 6)])
+        self.assertEqual(list(problem.subset_A), [2, 3])
+        self.assertEqual(list(problem.subset_B), [(1, 2), (5, 6)])
+        self.assertEqual(list(problem.C[1]), [-1, 3])
+        self.assertEqual(list(problem.C[2]), [4, 7])
+        self.assertEqual(list(problem.C[3]), [3, 8])
+        self.assertEqual(list(problem.D[(1, 2)]), [1, 5])
+        self.assertEqual(list(problem.D[(3, 4)]), [3])
+        self.assertEqual(list(problem.D[(5, 6)]), [6, 8, 9])
+        self.assertEqual(list(problem.subset_C[1]), [-1])
+        self.assertEqual(list(problem.subset_C[2]), [4])
+        self.assertEqual(list(problem.subset_C[3]), [3, 8])
+        self.assertEqual(list(problem.subset_D[(1, 2)]), [1, 5])
+        self.assertEqual(list(problem.subset_D[(3, 4)]), [])
+        self.assertEqual(list(problem.subset_D[(5, 6)]), [6])
+        self.assertEqual(list(problem.E[0]), [1, 2])
+        self.assertEqual(list(problem.E[1]), [3])
+        self.assertEqual(list(problem.F[(1, 2, 3)]), [(1, 2)])
+        self.assertEqual(list(problem.F[(4, 5, 6)]), [(3, 4)])
+
+        # try adding elements to test the domains (1 compatible, 1 incompatible)
+        # set subset_A
+        problem.subset_A.add(1)
+        error_raised = False
+        try:
+            problem.subset_A.add(4)
+        except ValueError:
+            error_raised = True
+        self.assertEqual(error_raised, True)
```

Comment:
I realize you are copying a structure that is used elsewhere, but we have been (slowly) away from that and instead to make proper use of unittest.  Can you update these tests to use `assertRaisesRegex`:
```python
with self.assertRaisesRegex(ValueError, "message that we expect the exception to contain"):
    problem.subset_A.add(4)
```